File: DKImagePickerControllerDemo_OC/DKImagePickerControllerDemo_OC/CustomUIDelegate/CustomUIDelegate.m
//
//  CustomUIDelegate.m
//  DKImagePickerControllerDemo_OC
//
//  Created by zm on 2017/7/7.
//  Copyright © 2017年 zm. All rights reserved.
//

#import "CustomUIDelegate.h"
#import "CustomGroupDetailImageCell.h"
#import "CustomGroupDetailCameraCell.h"
@interface CustomUIDelegate()

@property (nonatomic, strong) UIToolbar * footer;

@end


@implementation CustomUIDelegate
- (UIToolbar *)footer{
    if (!_footer) {
        _footer = [[UIToolbar alloc] initWithFrame:CGRectMake(0, 0, 0, 44)];
        _footer.translucent = NO;
        _footer.items = @[[[UIBarButtonItem alloc] initWithBarButtonSystemItem:UIBarButtonSystemItemFlexibleSpace target:nil action:nil], [[UIBarButtonItem alloc] initWithCustomView:[self createDoneButtonIfNeeded]]];
        [self updateDoneButtonTitle:[self createDoneButtonIfNeeded]];
    }
    return _footer;
}



- (UIButton *)createDoneButtonIfNeeded{
    if (!self.doneButton) {
        self.doneButton = [UIButton buttonWithType:UIButtonTypeCustom];
        self.doneButton.titleLabel.font = [UIFont systemFontOfSize:15];
        [self.doneButton setTitleColor:[UIColor colorWithRed:85/255 green:184/255 blue:44/255 alpha:1.0] forState:UIControlStateNormal];
        [self.doneButton setTitleColor:[UIColor colorWithRed:85/255 green:183/255 blue:44/255 alpha:0.4] forState:UIControlStateDisabled];
        [self.doneButton addTarget:self.imagePickerController action:@selector(done) forControlEvents:UIControlEventTouchUpInside];
    }
    return self.doneButton;
}

- (void)prepareLayout:(DKImagePickerController *)imagePickerController vc:(UIViewController *)vc{
    self.imagePickerController = imagePickerController;
}
- (void)imagePickerController:(DKImagePickerController *)imagePickerController showsCancelButtonForVC:(UIViewController *)vc{
    vc.navigationItem.rightBarButtonItem = [[UIBarButtonItem alloc] initWithBarButtonSystemItem:UIBarButtonSystemItemCancel target:imagePickerController action:@selector(dismiss)];
}
- (void)imagePickerController:(DKImagePickerController *)imagePickerController hidesCancelButtonForVC:(UIViewController *)vc    {
    vc.navigationItem.rightBarButtonItem = nil;
}
- (UIView *)imagePickerControllerFooterView:(DKImagePickerController *)imagePickerController{
    return self.footer;
}

- (void)updateDoneButtonTitle:(UIButton *)button{
    if (self.imagePickerController.selectedAssets.count > 0) {
        [button setTitle:[NSString stringWithFormat:@"Send(%lu)", (unsigned long)self.imagePickerController.selectedAssets.count] forState:UIControlStateNormal];
        button.enabled = YES;
    }else{
        [button setTitle:@"Send" forState:UIControlStateNormal];
        button.enabled = NO;
    }
    [button sizeToFit];
    
}

- (Class)imagePickerControllerCollectionCameraCell{
    return [CustomGroupDetailCameraCell class];
}

- (Class)imagePickerControllerCollectionImageCell{
    return [CustomGroupDetailImageCell class];
}


@end

File: DKImagePickerControllerDemo/CustomUIDelegate/CustomUIDelegate.swift
//
//  CustomUIDelegate.swift
//  DKImagePickerControllerDemo
//
//  Created by ZhangAo on 03/01/2017.
//  Copyright © 2017 ZhangAo. All rights reserved.
//

import UIKit

open class CustomUIDelegate: DKImagePickerControllerDefaultUIDelegate {
    
    lazy var footer: UIToolbar = {
        let toolbar = UIToolbar(frame: CGRect(x: 0, y: 0, width: 0, height: 44))
        toolbar.isTranslucent = false
        toolbar.items = [
            UIBarButtonItem(barButtonSystemItem: .flexibleSpace, target: nil, action: nil),
            UIBarButtonItem(customView: self.createDoneButtonIfNeeded()),
        ]
        self.updateDoneButtonTitle(self.createDoneButtonIfNeeded())
        
        return toolbar
    }()
    
    override open func createDoneButtonIfNeeded() -> UIButton {
        if self.doneButton == nil {
            let button = UIButton(type: .custom)
            button.titleLabel?.font = UIFont.systemFont(ofSize: 15)
            button.setTitleColor(UIColor(red: 85 / 255.0, green: 184 / 255.0, blue: 44 / 255.0, alpha: 1.0), for: .normal)
            button.setTitleColor(UIColor(red: 85 / 255.0, green: 184 / 255.0, blue: 44 / 255.0, alpha: 0.4), for: .disabled)
            button.addTarget(self.imagePickerController, action: #selector(DKImagePickerController.done), for: .touchUpInside)
            self.doneButton = button
        }
        
        return self.doneButton!
    }
    
    override open func prepareLayout(_ imagePickerController: DKImagePickerController, vc: UIViewController) {
        self.imagePickerController = imagePickerController
    }
    
    override open func imagePickerController(_ imagePickerController: DKImagePickerController,
                                               showsCancelButtonForVC vc: UIViewController) {
        vc.navigationItem.rightBarButtonItem = UIBarButtonItem(barButtonSystemItem: .cancel,
                                                               target: imagePickerController,
                                                               action: #selector(imagePickerController.dismiss as () -> Void))
    }
    
    override open func imagePickerController(_ imagePickerController: DKImagePickerController,
                                               hidesCancelButtonForVC vc: UIViewController) {
        vc.navigationItem.rightBarButtonItem = nil
    }
    
    override open func imagePickerControllerFooterView(_ imagePickerController: DKImagePickerController) -> UIView? {
        return self.footer
    }
    
    override open func updateDoneButtonTitle(_ button: UIButton) {
        if self.imagePickerController.selectedAssets.count > 0 {
            button.setTitle(String(format: "Send(%d)", self.imagePickerController.selectedAssets.count), for: .normal)
            button.isEnabled = true
        } else {
            button.setTitle("Send", for: .normal)
            button.isEnabled = false
        }
        
        button.sizeToFit()
    }
    
    open override func imagePickerControllerCollectionImageCell() -> DKAssetGroupDetailBaseCell.Type {
        return CustomGroupDetailImageCell.self
    }
    
    open override func imagePickerControllerCollectionCameraCell() -> DKAssetGroupDetailBaseCell.Type {
        return CustomGroupDetailCameraCell.self
    }

}
